packages/clerk-js/src/ui/components/OrganizationProfile/DomainList.tsx

```diff
@@ -21,6 +21,55 @@ type DomainListProps = GetDomainsParams & {
   fallback?: React.ReactNode;
 };
 
+const useDomainList = () => {
+  const { isAuthorizedUser: canDeleteDomain } = useGate({ permission: 'org:sys_domains:delete' });
+  const { isAuthorizedUser: canVerifyDomain } = useGate({ permission: 'org:sys_domains:manage' });
+
+  return {
+    showDotMenu: canDeleteDomain || canVerifyDomain,
+    canVerifyDomain,
+    canDeleteDomain,
+  };
+};
+
+const DomainListDotMenu = ({
+  redirectSubPath,
+  domainId,
+}: Pick<DomainListProps, 'redirectSubPath'> & {
+  domainId: OrganizationDomainResource['id'];
+}) => {
+  const { navigate } = useRouter();
+  const { canDeleteDomain, canVerifyDomain } = useDomainList();
+
+  return (
+    <ThreeDotsMenu
+      actions={[
+        ...(canVerifyDomain
+          ? [
+              {
+                label: localizationKeys(
+                  'organizationProfile.profilePage.domainSection.unverifiedDomain_menuAction__verify',
+                ),
+                onClick: () => navigate(`${redirectSubPath}${domainId}/verify`),
```

Comment:
🔧 Prefer the `buildUrl` or `buildPath` utilities that ensure the path is formed correctly without accidental double slashes. 

String concatenations in URLs should be avoided.

Reply:
I went with a custom solution, seems like buildUrl and the hashPath did not work with path routing for internal navigations